lib/cron/synth-switch.ts

```diff
@@ -41,14 +79,170 @@ const handler: ScheduledHandler = async (_event: EventBridgeEvent<string, void>)
+  const configs = validateConfigs(await readTokenConfig(log));
+
+  // We can't pass in arrays as parameters to the query, so we have to build it into a formatted string
+  // inputToken and outputToken MUST be sanitized and lowercased before being passed into the query
+  const tokenInList = "('" + configs.map((config) => config.inputToken).join("', '") + "')";
+  const tokenOutList = "('" + configs.map((config) => config.outputToken).join("', '") + "')";
+
+  log.info(
+    {
+      tokenInList,
+      tokenOutList,
+      valueTokenInList: String(tokenInList),
+      valueTokenOutList: String(tokenOutList),
+    },
+    'formatted tokenInList, tokenOutList'
+  );
+
+  async function updateSynthSwitchRepository(configs: TokenConfig[], result: ResultRowType[]) {
+    // match configs to results
+    const configMap: {
+      [key: string]: ResultRowType[];
+    } = {};
+    result.map((row) => {
+      const key = `${row.tokenin}#${row.tokeninchainid}#${row.tokenout}#${row.tokenoutchainid}`;
+      if (key in configMap) {
+        configMap[key].push(row);
+      } else {
+        configMap[key] = [row];
+      }
+    });
+
+    for (const config of configs) {
+      // totalTrades is both ExactIn and ExactOut
+      const ordersForConfig =
+        configMap[
+          `${config.inputToken}#${config.inputTokenChainId}#${config.outputToken}#${config.outputTokenChainId}`
+        ];
+      // build trade objects differentiating between ExactIn and ExactOut
+      let tradeOutcomesByKey: {
+        [key: string]: {
+          pos: number;
+          neg: number;
+        };
+      } = {};
+      for (const order of ordersForConfig) {
+        const tradeType =
+          order.classic_amountin == order.classic_amountingasadjusted ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT;
+        const trade: SynthSwitchRequestBody = {
+          inputToken: order.tokenin,
+          inputTokenChainId: Number(order.tokeninchainid),
+          outputToken: order.tokenout,
+          outputTokenChainId: Number(order.tokenoutchainid),
+          type: String(tradeType),
+          // classic amount in here or synthetic amount in?
+          amount: tradeType == TradeType.EXACT_INPUT ? order.classic_amountin : order.classic_amountout,
+        };
+        const key = SwitchRepository.getKey(trade);
+        let hasPriceImprovement: boolean;
+        if (tradeType == TradeType.EXACT_INPUT) {
+          hasPriceImprovement = BigNumber.from(order.settledAmountOut).gt(order.classic_amountoutgasadjusted);
+        } else {
+          hasPriceImprovement = BigNumber.from(order.classic_amountingasadjusted).gt(order.settledAmountIn);
+        }
+
+        if (!(key in tradeOutcomesByKey)) {
+          tradeOutcomesByKey[key] = {
+            pos: 0,
+            neg: 0,
+          };
+        }
+        if (hasPriceImprovement) {
+          tradeOutcomesByKey[key].pos++;
+        } else {
+          tradeOutcomesByKey[key].neg++;
+        }
+      }
```

Comment:
nit: might be nice to pull this logic out into another function something like `hasPositiveTradeOutcome()` then we can add any kind of arbitrary logic into that function to determine if trade was favorable. Mainly thinking about this if we wanted to add some kind of check for trade duration.

Reply:
oo yeah good idea